c/tskit/tables.c

```diff
@@ -1418,11 +1431,13 @@ tsk_edge_table_copy(tsk_edge_table_t *self, tsk_edge_table_t *dest, tsk_flags_t
     int ret = 0;
 
     if (!(options & TSK_NO_INIT)) {
-        ret = tsk_edge_table_init(dest, 0);
+        ret = tsk_edge_table_init(dest, self->options);
```

Comment:
I did this to avoid the case of turning metadata back on for a table that was init'd with it off. I've added an error for that case so that now you can disable metadata on a copy. https://github.com/tskit-dev/tskit/pull/712/commits/071d93b71a71f54807b030aef958782d38cc37af